### ISSUE TITLE
Unable to verify JWT claims

### ISSUE THREAD

**Issue by Linkaara** [jumbojett/OpenID-Connect-PHP#145]
Hello
I am trying to use the client_example.php provided for a basic sign-in function, but I am getting a "Unable to verify JWT claims"
This is my code:

```
require __DIR__ . '/testC/vendor/autoload.php';

use Jumbojett\OpenIDConnectClient;
try {
$oidc = new OpenIDConnectClient('https://login.microsoftonline.com/common',    'xxxxxxx-xxxxxxx-xxxxxx-xxx',    'xxxxxxxx');

$oidc->authenticate();
$name = $oidc->requestUserInfo('given_name');
} catch (Exception $e) {
    echo 'Caught exception: ',  $e->getMessage(), "\n";
}
```

I was also getting a  "AADSTS70002: Error validating credentials" also, which stopped when I changed the special characters on the client secret to ascii characters (= to %3D  for example)

Thanks!

**Comment by jumbojett** [comment]
@Linkaara have you seen _Example 6_ in the README? It might be relevant since you're authenticating against microsoft. Let us know if this helps.

**Comment by findus** [comment]
I get the same error trying to implement OIDC at Microsoft Azure AD, but I found the reason:

Using the well-known config of the common endpoint, the valid issuer url contains a placeholder for the tenant id.
So the following matching in verifyJWTClaims always evaluates to false:
`return (($claims->iss == $this->getIssuer() || $claims->iss == $this->getWellKnownIssuer() || $claims->iss == $this->getWellKnownIssuer(true))`

Some other libraries support wildcards but that won't help in this case.

I suggest to extend the issuer check to allow a custom check method (e.g. lambda function) to be passed to the client.

**Comment by jumbojett** [comment]
Fixed in #166
